### PR TITLE
APEXCORE-769 Clearing oioGroups to remove information from prior depl…

### PR DESCRIPTION
**Pull request by francisf**
…oyments

@tushargosavi please review

### CODE REVIEW

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java [hunk 1]
```diff
@@ -1290,6 +1290,7 @@ private void deployInputStreams(List<OperatorDeployInfo> operatorList, HashMap<S
    */
   private void setupOiOGroups(Map<Integer, Integer> oioNodes)
   {
+    oioGroups.clear(); //Reset prior info if any before redeployment
```
Comment (vrozov) [inline engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java]: should not clean up be handled in undeploy()? What if AM decides to deploy another operator/partition to a container that has few operators with thread local stream deployed? Will not that lead to the existing OiO groups being lost?

**Review by vrozov — CHANGES_REQUESTED** [review]

**Comment by francisf** [comment]
@vrozov @tushargosavi  please review

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java (hunk 1 above)
Comment (francisf) [reply engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java]: @vrozov I've done the changes. Please review.

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java [hunk 2]
```diff
@@ -143,6 +143,8 @@
    * value: list of nodes which are in oio with oio owning thread node
    */
   protected final Map<Integer, ArrayList<Integer>> oioGroups = new ConcurrentHashMap<>();
+  //a simple map which maps the oio node to it's the node which owns the thread.
+  protected final Map<Integer, Integer> oioNodes = new ConcurrentHashMap<>();
```
Comment (tushargosavi) [inline engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java:147]: As you have made it global make sure that the entries are consistent with oioGroups. i.e the parent to children mapping and child to parent mapping is always consistent in both the structures.

File: engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java [hunk 3]
```diff
@@ -566,6 +568,11 @@ private synchronized void undeploy(List<Integer> nodeList)
 
     for (Integer operatorId : nodeList) {
       nodes.remove(operatorId);
+      Integer sourceNodeId = oioNodes.get(operatorId);
+      if (sourceNodeId != null) {
+        ArrayList<Integer> oioNodeIdList = oioGroups.get(sourceNodeId);
```
Comment (tushargosavi) [inline engine/src/main/java/com/datatorrent/stram/engine/StreamingContainer.java:573]: I feel the problem could be solve by changing list to set.

**Comment by vrozov** [comment]
@francisf @tushargosavi closed due to inactivity.

**Pull request closed without merging**